software/technical-requirements.qmd

```diff
@@ -0,0 +1,108 @@
+-   For building the web and API layer, we will use **Django** for the
+    framework since it is very popular and has a large amount of online
+    resources. Django also implements **RESTful API** (representational
+    state transfer), which is the de facto standard way for building
+    APIs.
+
+For our decisions to use these technologies, see [Decision
+Record](https://decisions.seedcase-project.org).
+
+## Technical constraints
+
+Following our guiding principles, we have the following technical
+constraints of the Seedcase framework:
+
+| Constraints                                         | Background/Motivation                                                            |
+|----------------------------|--------------------------------------------|
+| Run on Windows, MacOS, and Linux                    | Our potential users work on any of these systems, so we need to design for that  |
+| Use open source software dependencies               | We will use an open license, so we need to use components that are open as well  |
+| Use software that's (relatively) familiar to many   | We want others to contribute, so we need to use tools others (likely) know       |
+| Integrate GDPR, privacy, and security compliance    | Our target users work with health data, so this is vital to consider             |
+| Deployable to servers and locally                   | Could be used locally but mainly used on a server environment                    |
+| Storage and computing may be at different locations | Where data are stored vs analyzed will likely be different, see subsection below |
+
+: Technical constraints we need to consider for Seedcase software.
+
+## Conventions
+
+We follow the conventions below for software development when developing
+the Seedcase framework:
+
+| Conventions                                                                                                                | Background/Motivation                                                                                                                   |
+|-------------------------------|-----------------------------------------|
+| Follow standard styling (e.g., [PIP3](https://pypi.org/project/pip/) and [black](https://github.com/psf/black) formatting) | Coding style is vital to readability, so we follow best practices                                                                       |
+| Use [CalVer](https://calver.org/) versioning                                                                               | This style is simple and easily shows the age of the software                                                                           |
+| Incorporate test-driven and documentation-driven development styles                                                        | Each has pros and cons, we use what works best. See the [Contributing Guidelines](https://seedcase-project.org/community/contributing). |
+
+: Conventions that we follow for the Seedcase framework.
```

Comment:
This section could be moved to the Style documentation?

Reply:
Good idea. Once it is setup more, we can move it over ☺️

Reply:
Alright. Will leave it here for now then